Community Toolbox/scripts/utils_CommunityToolboxDraw/utils_CommunityToolboxDraw.gml

```diff
@@ -0,0 +1,266 @@
+function draw_set_color_alpha(_color, _alpha) {
+    gml_pragma("forceinline");
+    draw_set_color(_color);
+    draw_set_alpha(_alpha);
+}
+
+/// @func draw_set_colour_alpha(colour,alpha)
+/// @desc Sets the drawing colour and alpha.
+/// @arg {Constant.Color} colour    The drawing colour to set.
+/// @arg {Real} alpha               The alpha to set (between 0 and 1).
+function draw_set_colour_alpha(_colour, _alpha) {
+    gml_pragma("forceinline");
+    draw_set_colour(_colour);
+    draw_set_alpha(_alpha);
+}
+
+/// @func draw_set_align(halign,valign)
+/// @desc Sets the horizontal and vertical alignment of the drawn text.
+/// @arg {Constant.HAlign} halign       The horizontal alignment to set.
+/// @arg {Constant.VAlign} valign       The vertical alignment to set.
+function draw_set_align(_halign, _valign) {
+    gml_pragma("forceinline");
+    draw_set_halign(_halign);
+    draw_set_valign(_valign);
+}
+
+#endregion
+
+#region Shapes
+
+/// @func draw_arc(x,y,radius,anglefrom,angleto,[precision])
+/// @desc Draws a circular line around the given center, with the given radius and between given angles.
+/// @arg {Real} x               The x coordinate of the arc center.
+/// @arg {Real} y               The y coordinate of the arc center.
+/// @arg {Real} radius          The radius of the arc around the center.
+/// @arg {Real} anglefrom       The starting angle of the arc.
+/// @arg {Real} angleto         The ending angle of the arc.
+/// @arg {Real} [precision]     The precision of the circle the arc is a part of.
+function draw_arc(_x, _y, _radius, _anglefrom, _angleto, _precision = 24) {
+    // static functions    
+    static get_segment_position = function(_angle, _segment_angle) {
+        _angle = _angle - floor(_angle / _segment_angle) * _segment_angle;
+        if (_angle == 0)
+            return 0;
+        
+        var _yfrom = dsin(_angle);
+        var _yto = dsin(_angle - _segment_angle);
+        return _yfrom / (_yfrom - _yto);
+    }
+    
+    static draw_circle_vertex = function(_x, _y, _radius, _angle) {
+        draw_vertex(_x + lengthdir_x(_radius, _angle), _y + lengthdir_y(_radius, _angle));
+    }
+    
+    static draw_midsegment_vertex = function(_x, _y, _radius, _angle1, _angle2, _amount) {
+        if (_amount == 0) {
+            draw_circle_vertex(_x, _y, _radius, _angle1);
+            return;
+        }
+        
+        var _x1 = _x + lengthdir_x(_radius, _angle1);
+        var _y1 = _y + lengthdir_y(_radius, _angle1);
+        var _x2 = _x + lengthdir_x(_radius, _angle2);
+        var _y2 = _y + lengthdir_y(_radius, _angle2);
+        draw_vertex(lerp(_x1, _x2, _amount), lerp(_y1, _y2, _amount));
+    }
+    
+    static normalize_angles = function(_anglefrom, _angleto) {
+        static result = [0, 0];
+        
+        result[0] = min(_anglefrom, _angleto);
+        result[1] = max(_anglefrom, _angleto);
+        
+        if (result[0] + 360 < result[1]) {
```

Comment:
Is the behavior of these functions well-defined when `anglefrom > angleto`? I would expect it to just draw the arc in reverse. If that behavior is well-defined, then I think you need a case for `result[1] + 360 < result[0]` here as well. If not, that's probably fine too, as long as it's documented.

Reply:
Drawing arc from A to B and from B to A should yield the same results; the angles normalisation a few lines before ensures that `result[0]` is no higher than `result[1]`, so there's no way `result[1] + 360 < result[0]` can ever happen.

Reply:
Ah, I guess my eyes glazed over the min/max bit. That makes sense